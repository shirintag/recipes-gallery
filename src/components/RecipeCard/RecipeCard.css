.recipe-card {
  width: 260px;
  margin: 20px;
  padding: 20px;
  border: 1px solid #e8e8e8;
  border-radius: 3px;
  box-sizing: border-box;
  background-color: white;
}

.recipe-image {
  width: 220px;
  height: 130px;
  margin-bottom: 16px;
}

.recipe-data {
  text-align: left;
}

span {
  margin-right: 10px;
}
