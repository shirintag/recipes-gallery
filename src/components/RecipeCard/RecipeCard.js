import React from "react";
import StarRating from "components/Rater/StarRating";
import Heart from "components/Heart/Heart";

import "./RecipeCard.css";
const RecipeCard = (props) => {
  return (
    <div className="recipe-card">
      <div className="recipe-heart">
        <Heart />
      </div>
      <div className="recipe-image-container">
        <img
          className="recipe-image"
          src={
            "https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
          }
          alt={props.name}
        />
      </div>

      <div className="recipe-data f-subcopy">
        <StarRating totalStars={5} />
        <div className="f-title">{props.name}</div>
        <div className="f-subtitle">{props.headline}</div>
        <span>calories: {props.calories || "--"}</span>
        <span>Time: {props.time.substring(2) || "--"}</span>
      </div>
    </div>
  );
};

export default RecipeCard;
