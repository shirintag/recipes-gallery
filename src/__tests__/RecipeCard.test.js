import React from "react";
import { render } from "@testing-library/react";

import RecipeCard from "../components/RecipeCard/RecipeCard";
const RecipeData = {
  calories: "516 kcal",
  headline: "with Sweet Potato Wedges and Minted Snap Peas",
  id: "533143aaff604d567f8b4571",
  image:
    "https://d3hvwccx09j84u.cloudfront.net/web/image/533143aaff604d567f8b4571.jpg",
  name: "Crispy Fish Goujons ",
  time: "PT35M",
};

describe("ProductCard", () => {
  it("should render", () => {
    const { container } = render(<RecipeCard {...RecipeData} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
