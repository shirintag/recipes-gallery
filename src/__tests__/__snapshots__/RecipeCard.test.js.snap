// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render 1`] = `
<div
  class="recipe-card"
>
  <div
    class="recipe-heart"
  >
    <div>
      <div
        class="heart"
      />
    </div>
  </div>
  <div
    class="recipe-image-container"
  >
    <img
      alt="Crispy Fish Goujons "
      class="recipe-image"
      src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
    />
  </div>
  <div
    class="recipe-data f-subcopy"
  >
    <div
      class="star-rating"
    >
      <div
        class="star-container"
      >
        <div
          class="star"
        />
      </div>
      <div
        class="star-container"
      >
        <div
          class="star"
        />
      </div>
      <div
        class="star-container"
      >
        <div
          class="star"
        />
      </div>
      <div
        class="star-container"
      >
        <div
          class="star"
        />
      </div>
      <div
        class="star-container"
      >
        <div
          class="star"
        />
      </div>
    </div>
    <div
      class="f-title"
    >
      Crispy Fish Goujons 
    </div>
    <div
      class="f-subtitle"
    >
      with Sweet Potato Wedges and Minted Snap Peas
    </div>
    <span>
      Calories: 
      516 kcal
    </span>
    <span>
      Time: 
      35M
    </span>
  </div>
</div>
`;
